### ISSUE TITLE
Some @typescript-eslint rules doesn't work in vue files

### ISSUE THREAD

**Issue by trygveaa** [vuejs/vue-eslint-parser#40]
Some rules in @typescript-eslint check if the file name ends with `.ts` to determine if the rule should be run or not, because they don't apply to `.js` files.

For example, if I have this code:

```
const test = x => x;
```

If I place it in a `.ts` file, I will get a warning from `@typescript-eslint/explicit-function-return-type`. However, if I place it in a `<script lang="ts">` block of a `.vue` file, this warning does not appear. This is because @typescript-eslint checks the file ending, and since it is `.vue` and not `.ts`, the rule isn't run.

Is there some other way @typescript-eslint could check if the code is typescript, other than by the file extension? How can it differentiate between `.vue` files which specify `lang="ts"` and those that don't?

Here is where @typescript-eslint does this check:
https://github.com/typescript-eslint/typescript-eslint/blob/7be56570de9d1983af95db5900ec730930d52712/packages/eslint-plugin/src/rules/explicit-function-return-type.ts#L83
https://github.com/typescript-eslint/typescript-eslint/blob/7be56570de9d1983af95db5900ec730930d52712/packages/eslint-plugin/src/util/misc.ts#L11-L13

**Comment by mysticatea** [comment]
Thank you for this report.

This is a tough problem.
Parsers cannot modify file names for rules, so this parser cannot solve the problem.

Hmm.

Possibly, this parser can tell the typescript parser (instead of rules) something. We need to discussion with typescript-eslint team.

**Comment by seibert-io** [comment]
Hey @mysticatea,

thanks for looking into this. Just checking in to see if anyone had a chance to discuss this with the typescript-eslint team already?

**Comment by IlCallo** [comment]
The `.ts` extension check is gonna be removed into typescript-eslint in v2.0.0
https://github.com/typescript-eslint/typescript-eslint/pull/594

**Comment by IlCallo** [comment]
This should be resolved with `typescript-eslint` v2 and can probably be closed

**Comment by ota-meshi** [comment]
I close this issue. If you have any problems, please open a new issue or follow another related issue.
